rts/lra/coordinator/src/main/java/io/narayana/lra/coordinator/domain/model/LongRunningAction.java

```diff
@@ -109,6 +112,22 @@ public LongRunningAction(LRAService lraService, Uid rcvUid) {
         this.finishTime = null;
         this.status = LRAStatus.Active;
         this.scheduler = Executors.newScheduledThreadPool(1);
+        this.isFailedLRAType = false;
+    }
+
+    public LongRunningAction(LRAService lraService, Uid rcvUid, final boolean isFailed) {
+        super(rcvUid);
+
+        this.uid = rcvUid.fileStringForm();
+        this.lraService = lraService;
+        this.inFlight = false;
+        this.id = null;
+        this.parentId = null;
+        this.clientId = null;
+        this.finishTime = null;
+        this.status = LRAStatus.Active;
+        this.scheduler = Executors.newScheduledThreadPool(1);
+        this.isFailedLRAType = isFailed;
```

Comment:
When the new constructor is needed, could be reused the initialization code from the existing constructor where `this(...)` would be used for the fields already handled by the existing one?

Reply:
Since `isFailedLRAType` is final now, cannot use `this(...)` as it tries to set it's value twice, which would be a compilation error.

Reply:
Try this
```
    public LongRunningAction(LRAService lraService, Uid rcvUid) {
        this(lraService, rcvUid, false);
    }
```
but to be honest maybe asking for the field to be final is not relevant when all other fields are non-final. But I assume they should be all final. Never mind. Rather to simplify the constructor and not having the field final. Finality of the all fields would be good to be reevaluated later probably.

Reply:
ok